web/ui/react-app/src/App.tsx

```diff
@@ -82,34 +83,34 @@ const App: FC<AppProps> = ({ consolesLink, agentMode }) => {
               React-handled router paths ("reactRouterPaths") in /web/web.go.
             */}
               <Route path="/agent">
-                <AgentPage />
+                <AgentPage staticReady={ready} />
```

Comment:
I'd try to avoid threading through such a minor setting to all pages... how about doing it like we do for the path prefix, which is also injected. We have a context for it here: https://github.com/prometheus/prometheus/blob/fa0f04bbc6f41708782d4e259fd1fb38896bb904/web/ui/react-app/src/contexts/PathPrefixContext.tsx#L4, which is wrapped around the whole App here: https://github.com/prometheus/prometheus/blob/fa0f04bbc6f41708782d4e259fd1fb38896bb904/web/ui/react-app/src/App.tsx#L74

And then any place that needs it can do `usePathPrefix()`.

Maybe at some point it can even make sense to combine it all into one big injected settings config context object, rather than multiple small ones.

Reply:
Done